src/core_modules/capture-core/components/Pages/Search/SearchPage.component.js

```diff
@@ -76,6 +76,8 @@ const Index = ({
         if (!preselectedProgram.value) {
             showInitialSearchPage();
         }
+
+        return () => showInitialSearchPage();
```

Comment:
This will solve the issue you mentioned. The one that when you come back to the back from the main page you still see the results displayed.